llvm/lib/Target/AMDGPU/SIISelLowering.cpp

```diff
@@ -15619,6 +15619,91 @@ void SITargetLowering::finalizeLowering(MachineFunction &MF) const {
     }
   }
 
+  // ISel inserts copy to regs for the successor PHIs
+  // at the BB end. We need to move the SI_WAVE_RECONVERGE right before the branch.
+  // Even we don't have to move SI_WAVE_RECONVERGE we need to take care of the
+  // S_CBRANCH_SCC0/1 as SI_WAVE_RECONVERGE overwrites SCC
```

Comment:
This should have been emitted as a copy pair to a temp vreg?

Reply:
Could you please clarify your question?